include/swift/Threading/Impl/Linux.h

```diff
@@ -134,6 +134,29 @@ inline void lazy_mutex_unsafe_unlock(lazy_mutex_handle &handle) {
   (void)::pthread_mutex_unlock(&handle);
 }
 
+// .. Recursive mutex support ................................................
+
+using recursive_mutex_handle = ::pthread_mutex_t;
+
+inline void recursive_mutex_init(mutex_handle &handle, bool checked = false) {
```

Comment:
It so happens that `mutex_handle` and `recursive_mutex_handle` are the same here, but really these should take `recursive_mutex_handle`.